glide-core/redis-rs/redis/tests/auth.rs

```diff
@@ -0,0 +1,310 @@
+                .query_async::<_, ()>(standalone_conn)
+                .await
+                .map(|_| ()),
+        }
+    }
+
+    async fn kill_connection(con: &mut ClusterConnection) {
+        let mut kill_cmd = cmd("client");
+        kill_cmd
+            .arg("kill")
+            .arg("type")
+            .arg("normal")
+            .arg("skipme")
+            .arg("yes");
+        con.route_command(&kill_cmd, ALL_SUCCESS_ROUTE)
+            .await
+            .unwrap();
+    }
+
+    async fn reset_connection_standalone(con: &mut MultiplexedConnection) {
+        let _: () = cmd("reset").query_async(con).await.unwrap();
+    }
+
+    #[tokio::test]
+    #[serial_test::serial]
+    async fn test_replace_password_cluster() {
+        let cluster_context = TestClusterContext::new(3, 0);
+
+        // Create a management connection to set the password
+        let mut management_connection =
+            match create_connection(None, ConnectionType::Cluster, Some(&cluster_context), None)
+                .await
+                .unwrap()
+            {
+                Connection::Cluster(conn) => conn,
+                _ => panic!("Expected ClusterConnection"),
+            };
+
+        // Set the password using the unified function
+        let mut management_conn = Connection::Cluster(management_connection.clone());
+        set_password(PASSWORD, &mut management_conn).await.unwrap();
+
+        // Test that we can't connect without password
+        let connection_should_fail =
+            create_connection(None, ConnectionType::Cluster, Some(&cluster_context), None).await;
+        assert!(connection_should_fail.is_err());
+
+        // Test that we can connect with password
+        let mut connection_should_succeed = match create_connection(
+            Some(PASSWORD.to_string()),
+            ConnectionType::Cluster,
+            Some(&cluster_context),
+            None,
+        )
+        .await
+        .unwrap()
+        {
+            Connection::Cluster(conn) => conn,
+            _ => panic!("Expected ClusterConnection"),
+        };
+
+        let res: RedisResult<Value> = cmd("set")
+            .arg("foo")
+            .arg("bar")
+            .query_async(&mut connection_should_succeed)
+            .await;
+        assert_eq!(res.unwrap(), Value::Okay);
+
+        // Verify that we can retrieve the set value
+        let res: RedisResult<Value> = cmd("get")
+            .arg("foo")
+            .query_async(&mut connection_should_succeed)
+            .await;
+        assert_eq!(res.unwrap(), Value::BulkString(b"bar".to_vec()));
+
+        // Kill the connection to force reconnection
+        kill_connection(&mut management_connection).await;
+
+        // Attempt to get the value again to ensure reconnection works
+        let should_be_ok: RedisResult<Value> = cmd("get")
+            .arg("foo")
+            .query_async(&mut connection_should_succeed)
+            .await;
+        assert!(should_be_ok.is_ok());
```

Comment:
why here you don't check the received value but in line 160 you do?

Reply:
No reason, both are fine, I don't test get. For the consistency, ill change.